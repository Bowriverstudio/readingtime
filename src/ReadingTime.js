/**
 * React / Frontity dependencies
 */
import PropTypes from "prop-types";
import React from "react";

/**
 * Internal dependencies
 */
import readingStats from "./ReadingStats";

/**
 * Estimates the Reading Time
 *
 * According to Medium, people read about 200 words per minute. Medium also adds 12 seconds for each inline image.
 *
 * Reference: https://medium.com/@nadeem4uwebtech/how-to-add-reading-time-in-wordpress-without-using-plugin-d2e8af7b1239
 */
const ReadingTime = ({
  theContent,
  wordsPerMinute = 200,
  imagesPerMinute = 5,
  className,
}) => {
  //   const stats = readingTime(theContent, options);
  const stats = readingStats(theContent, wordsPerMinute, imagesPerMinute);
  return <span className={className}>{`${stats.text}`}</span>;
};

ReadingTime.propTypes = {
  /**
   *  the_content of the WordPress Post.
   */
  theContent: PropTypes.string.isRequired,

  /**
   *  Words Per Minute read
   */
  wordsPerMinute: PropTypes.number,

  /**
   * Images per min viewed
   */
  imagesPerMinute: PropTypes.number,

  /**
   * Optional className for css
   */
  className: PropTypes.string,
};

export default ReadingTime;
