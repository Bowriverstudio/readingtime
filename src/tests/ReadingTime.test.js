/**
 * React / Frontity dependencies
 */
import React from "react";
import ReactDOM from "react-dom";

/**
 * Internal dependencies
 */
import ReadingTime from "../ReadingTime";

it("ReadingTime renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ReadingTime />, div);
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});
