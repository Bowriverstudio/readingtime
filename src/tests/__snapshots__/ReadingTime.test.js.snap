// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadingTime renders without crashing 1`] = `
<div>
  <span>
    0 mins
  </span>
</div>
`;
